### ISSUE TITLE
Enhance `Input Search` component with dropdown to visualize search results

### ISSUE THREAD

**Issue by o-hook** [porsche-design-system/porsche-design-system#4703]
### Scope

Currently the PDS `p-input-search` component doesn't provide an integrated dropdown to visualize search results items (e.g. a link lists). This pattern is very often used (e.g. for dealer searches) and teams must build the dropdown functionality by themselves which have accessibility defects due to cross-shadowDOM boundaries.

See also discussion in Slack Channel: https://porschedev.slack.com/archives/CT7AVHTTQ/p1788269625685739

<img width="552" height="424" alt="Image" src="https://github.com/user-attachments/assets/be4cd7e7-e07c-43e5-8773-b6978a371a5d" />

### Notes

**Probable solution**
Use `p-input search` and enhance it (with a prop) to render a dropdown where users can slot e.g. search result items.

**Information:** 
- Good article with different patterns: https://www.makethingsaccessible.com/guides/accessible-site-search-with-combobox-suggestions/#lets-build-some-examples
- Probably we should go with the "**The combobox and dialog approach**", but also the other solutions need to be evaluated

Figma Link:
https://www.figma.com/design/EkdP468u4ZVuIRwalKCscb/branch/iLZkUfgJJcHL6OY0buO25U/Porsche-Web-Design-System?m=auto&node-id=61456-8628&t=eL4hTr8UlIjJxXbl-1

### Acceptance criteria

- [ ] `p-input-search` component is enhanced with a dropdown functionality to provide visual feedback of search result items

**Comment by neuncreative** [comment]
I've added the Figma branch
